codex-rs/core/src/shell.rs

```diff
@@ -210,6 +327,93 @@ pub async fn default_user_shell() -> Shell {
     }
 }
 
+#[cfg(test)]
+mod prettify_command_for_display_tests {
+    use super::prettify_command_for_display;
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn cd_prefix_in_bash_script_is_hidden() {
+        let command = vec![
+            "bash".to_string(),
+            "-lc".to_string(),
+            "cd foo && echo hi".to_string(),
+        ];
+        let display = prettify_command_for_display(&command);
+        assert_eq!(
+            display,
+            vec!["bash".to_string(), "-lc".to_string(), "echo hi".to_string()]
+        );
+    }
+
+    #[test]
+    fn cd_prefix_with_quoted_path_is_hidden() {
+        let command = vec![
+            "bash".to_string(),
+            "-lc".to_string(),
+            "  cd 'foo bar' && ls".to_string(),
+        ];
+        let display = prettify_command_for_display(&command);
+        assert_eq!(
+            display,
+            vec!["bash".to_string(), "-lc".to_string(), "ls".to_string()]
+        );
+    }
+
+    #[test]
+    fn cd_prefix_without_shell_wrapper_is_hidden() {
+        let command = vec![
+            "cd".to_string(),
+            "foo".to_string(),
+            "&&".to_string(),
+            "git".to_string(),
+            "status".to_string(),
+        ];
+        let display = prettify_command_for_display(&command);
```

Comment:
This does not seem right / the contract of this function is murky.

This would not work:

```
exec(["cd", "foo", "&&", "git", "status"])
```

only this would:

```
exec(["bash", "-lc", "cd foo && git status"])
```